File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for (i, num) in enumerate(nums):
            temp = target - num
            if temp in map:
                return [map[temp], i]
            map[num] = i

        return []

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();
        for (i, num) in nums.iter().enumerate() {
            let mut temp = target - num;
            if let Some(index) = map.get(&temp) {
                return vec![*index as i32, i as i32];
            }
            map.insert(num, i);
        }
        vec![]
    }
}